R/zchunk_L2231.land_input_3_irr.R

```diff
@@ -0,0 +1,356 @@
+                          LN1 = "LandNode1", LN2 = "LandNode2", LN3 = "LandNode3") ->
+      L223.LC_bm2_R_Mgd3_Yh_GLU
+
+    # L223.LN3_HistMgdAllocation_noncrop: Historical land cover, managed land in the third nest (noncrop)
+    # L223.LN3_MgdAllocation_noncrop: Model base period land cover, managed land in the third nest (noncrop)
+    # Formed from filtering the master table by different years.
+    L223.LC_bm2_R_Mgd3_Yh_GLU %>%
+      filter(year %in% LAND_HISTORY_YEARS) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["LN3_HistMgdAllocation"]])) ->
+      L223.LN3_HistMgdAllocation_noncrop
+
+    L223.LC_bm2_R_Mgd3_Yh_GLU %>%
+      filter(year %in% BASE_YEARS) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["LN3_MgdAllocation"]])) ->
+      L223.LN3_MgdAllocation_noncrop
+
+
+    # L223.LN3_UnmgdCarbon: Carbon content info, unmanaged land in the third nest
+    L223.LC_bm2_R_Unmgd3_Yh_GLU %>%
+      filter(year == max(BASE_YEARS)) %>%
+      left_join_error_no_match(select(GCAMLandLeaf_CdensityLT, Land_Type, LandLeaf), by = c("Land_Type" = "LandLeaf")) %>%
+      rename(Cdensity_LT = Land_Type.y) %>%
+      add_carbon_info(carbon_info_table = L121.CarbonContent_kgm2_R_LT_GLU) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["LN3_UnmgdCarbon"]])) ->
+      L223.LN3_UnmgdCarbon
+
+    # If any regions are zero in all periods, they will return missing values here. The specific value doesn't matter because
+    # these land use types will be zero in all periods, but there is an assert in the code prohibiting mature age from being zero.
+    L223.LN3_UnmgdCarbon[is.na(L223.LN3_UnmgdCarbon)] <- 1
+
+
+   # L223.LN3_MgdCarbon_noncrop: Carbon content info, managed land in the third nest, non-crop (forest)
+    L223.LC_bm2_R_Mgd3_Yh_GLU %>%
+      filter(year == max(BASE_YEARS)) %>%
+      left_join_error_no_match(select(GCAMLandLeaf_CdensityLT, Land_Type, LandLeaf), by = c("Land_Type" = "LandLeaf")) %>%
+      rename(Cdensity_LT = Land_Type.y) %>%
+      add_carbon_info(carbon_info_table = L121.CarbonContent_kgm2_R_LT_GLU) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["LN3_MgdCarbon"]])) ->
+      L223.LN3_MgdCarbon_noncrop
+
+
+    # L223.LN3_NoEmissCarbon: Set the no-emiss-carbon-calc as the type of carbon to use in forest leaves.
+    # Set Forests to use node-carbon-calc at the node level and no-emiss-carbon-calc at the leaf
+    # to allow them to switch between each other without lots of emissions+uptake.
+    L223.LN3_UnmgdCarbon %>%
+      filter(grepl("Forest", UnmanagedLandLeaf)) %>%
+      select(region, LandAllocatorRoot, LandNode1, LandNode2, LandNode3, UnmanagedLandLeaf) %>%
+      rename(LandLeaf = UnmanagedLandLeaf) ->
+      L223.LN3_NoEmissCarbon
+
+    L223.LN3_MgdCarbon_noncrop %>%
+      select(region, LandAllocatorRoot, LandNode1, LandNode2, LandNode3, LandLeaf) %>%
+      bind_rows(L223.LN3_NoEmissCarbon, .) %>%
+      mutate(no.emiss.carbon.calc = " ",
+             extra = "junk") ->
```

Comment:
?

Reply:
No clue, I kept that from the original code; it seems so stupid that I assumed it had to be important for running gcam. At the very least, it has to stay in an `OLD_DATA_SYSTEM_BEHAVIOR` flag to pass tests.

Reply:
this is the comment from the old code, which I can (and should have) bring over:
```
# Just need to create an empty tag, to do so we have to have a whitespace column and an
# extra column with extraneous data
```